src/hrflow_connectors/core/auth.py

```diff
@@ -18,6 +18,22 @@ class NoAuth(Auth):
     pass
 
 
+class APIKeyAuth(Auth):
+    api_key: str
+
+    def update(
+        self,
+        params: Union[Dict[str, str], None] = None,
+        headers: Union[Dict[str, str], None] = None,
+        payload: Union[Dict[str, str], None] = None,
+        cookies: Union[Dict[str, str], None] = None,
+    ):
+        if headers is not None:
+            headers.update(
+                {"Authorization": self.api_key}
+            )
```

Comment:
Actually, it depends. 

At Hrflow.ai, we also use `APIKey` as Auth. However, the key is not `Authorization` but `X-API-KEY`. I would have tried to make something a bit more general that allows us to choose the name of the key. 
It can be used for `SmartRecruiter` but also for others. And you, what you have to do, is inherit from this class to override the `key="Authorization"` attribute.

Thus, we would have :
```python
class APIKeyAuth(Auth):
  key:str
  value:str
...

class AuthorizationKeyAuth(APIKeyAuth):
  key:str = "Authorization"
```